spec/spec_helper.rb

```diff
@@ -12,4 +12,18 @@
 
   # Fix encoding issue in Rails 5.0; allows passing empty arrays or hashes as params.
   config.before(:each, type: :controller) { @request.env["CONTENT_TYPE"] = 'application/json' }
+
+  # Setting this to true keeps the database clean by rolling back any changes.
+  config.use_transactional_fixtures = true
+
+  # Some tests don't work within a transaction. Then we use DatabaseCleaner.
+  config.before(:each, concurrency: true) do
+    config.use_transactional_fixtures = false
+    DatabaseCleaner.strategy = :deletion, { except: ['spree_countries', 'spree_states'] }
+    DatabaseCleaner.start
+  end
+  config.append_after(:each, concurrency: true) do
+    DatabaseCleaner.clean
+    config.use_transactional_fixtures = true
+  end
```

Comment:
This doesn't make sense to me. The same code is already in the base_spec_helper.rb which is loaded further up. Strangely, we still `require` the database_cleaner after that. Maybe we should remove that?

Reply:
Yes, agree - my wrong logic here. 

I've tested this again, and indeed, this change plays no role - closing this PR. I don't know what to try next as bisecting yielded no useful results...